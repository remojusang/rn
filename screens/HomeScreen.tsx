import React from 'react';
import type { NativeStackScreenProps } from '@react-navigation/native-stack';
import { RootStackParamList } from '../utils/types';
import ScreenLayout from '../components/ScreenLayout';
import PersonList from '../components/List';
import LangSelector from '../components/LangSelector';
import { View } from 'react-native';
import { PersonType } from '../utils/types';
import Person from '../components/ListItem/Person';
import { personDataState } from '../components/Atoms';
import FloatingBtn from '../components/FloatingBtn';
import { useRecoilState } from 'recoil';

type HomeProps = NativeStackScreenProps<RootStackParamList, 'Home'>;

function HomeScreen({}: HomeProps) {
  const personData = useRecoilState(personDataState);
  return (
    <ScreenLayout isLoading={false}>
      <View style={{ width: '100%', alignItems: 'flex-end' }}>
        <LangSelector />
      </View>
      <PersonList
        isRefreshing={false}
        onRefresh={() => {}}
        flatListConfig={{
          data: personData[0] as PersonType[],
          renderItem: props => <Person {...props} />,
          keyExtractor: item => item.name,
        }}
      />
      <FloatingBtn />
    </ScreenLayout>
  );
}

export default HomeScreen;
