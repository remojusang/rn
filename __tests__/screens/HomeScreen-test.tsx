import { render } from '@testing-library/react-native';
import HomeScreen from '../../screens/HomeScreen';

// Couldn't find a navigation object. Is your component inside NavigationContainer? 에러 방지 코드
// https://spin.atomicobject.com/2021/02/24/react-navigation-5-unit-testing-components/
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('HomeScreen', () => {
  test('렌더링 테스트', () => {
    const TEST_PROPS: any = {
      route: jest.fn(),
      navigation: jest.fn(),
    };
    const renderedJson = render(
      <HomeScreen {...TEST_PROPS} />,
    ).toJSON();
    expect(renderedJson).toMatchSnapshot();
    expect(renderedJson).toBeTruthy();
  });
});
