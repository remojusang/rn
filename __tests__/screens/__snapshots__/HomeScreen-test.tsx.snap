// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 렌더링 테스트 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(50,50,50,1)",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text>
    Home
  </Text>
  <View>
    <Text
      onPress={[Function]}
    >
      로그인
    </Text>
  </View>
  <View>
    <Text
      onPress={[Function]}
    >
      회원가입
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "age": 15,
          "gender": "Male",
          "name": "Kalid",
        },
        Object {
          "age": 20,
          "gender": "Male",
          "name": "Fomagran",
        },
        Object {
          "age": 19,
          "gender": "Female",
          "name": "May",
        },
        Object {
          "age": 29,
          "gender": "Female",
          "name": "Julia",
        },
        Object {
          "age": 18,
          "gender": "Male",
          "name": "Hunter",
        },
        Object {
          "age": 28,
          "gender": "Female",
          "name": "Jane",
        },
        Object {
          "age": 34,
          "gender": "Male",
          "name": "Kendrick",
        },
        Object {
          "age": 25,
          "gender": "Female",
          "name": "Kali",
        },
        Object {
          "age": 37,
          "gender": "Male",
          "name": "Jay",
        },
        Object {
          "age": 31,
          "gender": "Male",
          "name": "Mac",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Kalid
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            15
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Fomagran
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            20
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            May
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            19
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Female
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Julia
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            29
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Female
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Hunter
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            18
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Jane
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            28
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Female
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Kendrick
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            34
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Kali
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            25
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Female
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Jay
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            37
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(50,50,50,1)",
              "borderColor": "white",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 5,
              }
            }
          >
            Mac
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            31
             years old
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 15,
              }
            }
          >
            Male
             
            ♀
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
